pkg/cmd/extension/manager.go

```diff
@@ -829,3 +852,26 @@ func codesignBinary(binPath string) error {
 	cmd := exec.Command(codesignExe, "--sign", "-", "--force", "--preserve-metadata=entitlements,requirements,flags,runtime", binPath)
 	return cmd.Run()
 }
+
+// cleanExtensionUpdateDir deletes extension state directory to avoid problems reinstalling extensions.
+func (m *Manager) cleanExtensionUpdateDir(name string) error {
+	updatePath := m.UpdateDir(name)
+	if _, err := os.Stat(updatePath); err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to check previous extension update state: %w", err)
+	}
+	if err := os.RemoveAll(updatePath); err != nil {
```

Comment:
@williammartin mentioned that maybe this block isn't actually needed as `os.RemoveAll` can handle the directory being there or not